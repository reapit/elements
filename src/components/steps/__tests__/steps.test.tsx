import { render, fireEvent } from '@testing-library/react'
import { Steps, StepsProps, StepsVertical, StepsVerticalProps, handleStepClick } from '../index'

describe('Steps', () => {
  const props: StepsProps = {
    steps: ['1', '2', '3', '4', '5'],
    onStepClick: vi.fn(),
  }

  it('should match a snapshot', () => {
    expect(render(<Steps {...props} />)).toMatchSnapshot()
  })

  it('should match a snapshot when selectedStep is different', () => {
    expect(render(<Steps {...props} selectedStep="2" />)).toMatchSnapshot()
  })

  it('should fire the onStepClick event correctly', async () => {
    const wrapper = render(<Steps {...props} />)
    fireEvent.click(wrapper.getByTestId('step-1'))
    expect(props.onStepClick).toHaveBeenCalledTimes(1)
  })

  afterEach(() => {
    vi.resetAllMocks()
  })
})

describe('StepsVertical', () => {
  const props: StepsVerticalProps = {
    steps: [
      {
        item: '1',
        content: 'Lorem, ipsum dolor',
      },
      {
        item: '2',
        content: 'Lorem, ipsum dolor',
      },
      {
        item: '3',
        content: 'Lorem, ipsum dolor',
      },
      {
        item: '4',
        content: 'Lorem, ipsum dolor',
      },
      {
        item: '5',
        content: 'Lorem, ipsum dolor',
      },
    ],
    onStepClick: vi.fn(),
    selectedStep: '5',
  }

  it('should match a snapshot', () => {
    expect(render(<StepsVertical {...props} />)).toMatchSnapshot()
  })

  it('should match a snapshot when selectedStep is different', () => {
    expect(render(<StepsVertical {...props} selectedStep="2" />)).toMatchSnapshot()
  })

  it('should fire the onStepClick event correctly', async () => {
    const wrapper = render(<StepsVertical {...props} />)

    fireEvent.click(wrapper.getByTestId('step-1'))

    expect(props.onStepClick).toHaveBeenCalledTimes(1)
  })

  afterEach(() => {
    vi.resetAllMocks()
  })
})

describe('handleStepClick', () => {
  it('should call onStepClick with the correct argument', () => {
    const onStepClick = vi.fn()
    const step = 'step1'

    const curried = handleStepClick(step, onStepClick)

    curried()

    expect(onStepClick).toHaveBeenCalledWith(step)
  })
})
