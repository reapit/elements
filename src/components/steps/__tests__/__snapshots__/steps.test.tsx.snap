// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Steps > should match a snapshot 1`] = `
<DocumentFragment>
  <div
    class="mocked-styled-0 el-steps"
  >
    <div
      class="mocked-styled-1 el-step"
      data-testid="step-0"
      role="button"
      tabindex="0"
    >
      1
    </div>
    <div
      class="mocked-styled-1 el-step"
      data-testid="step-1"
      role="button"
      tabindex="0"
    >
      2
    </div>
    <div
      class="mocked-styled-1 el-step"
      data-testid="step-2"
      role="button"
      tabindex="0"
    >
      3
    </div>
    <div
      class="mocked-styled-1 el-step"
      data-testid="step-3"
      role="button"
      tabindex="0"
    >
      4
    </div>
    <div
      class="mocked-styled-1 el-step"
      data-testid="step-4"
      role="button"
      tabindex="0"
    >
      5
    </div>
  </div>
</DocumentFragment>
`;

exports[`Steps > should match a snapshot when selectedStep is different 1`] = `
<DocumentFragment>
  <div
    class="mocked-styled-0 el-steps"
  >
    <div
      class="el-is-used el-step"
      data-testid="step-0"
      role="button"
      tabindex="0"
    >
      1
    </div>
    <div
      class="el-is-active el-step"
      data-testid="step-1"
      role="button"
      tabindex="0"
    >
      2
    </div>
    <div
      class="mocked-styled-1 el-step"
      data-testid="step-2"
      role="button"
      tabindex="0"
    >
      3
    </div>
    <div
      class="mocked-styled-1 el-step"
      data-testid="step-3"
      role="button"
      tabindex="0"
    >
      4
    </div>
    <div
      class="mocked-styled-1 el-step"
      data-testid="step-4"
      role="button"
      tabindex="0"
    >
      5
    </div>
  </div>
</DocumentFragment>
`;

exports[`StepsVertical > should match a snapshot 1`] = `
<DocumentFragment>
  <div
    class="mocked-styled-2 el-steps-vertical"
  >
    <div
      class="mocked-styled-3 el-step-vertical"
    >
      <div
        class="mocked-styled-4 el-step-vertical-item"
      >
        <div
          class="mocked-styled-1 el-step"
          data-testid="step-0"
          role="button"
          tabindex="0"
        >
          1
        </div>
      </div>
      <div
        class="mocked-styled-5 el-step-vertical-content"
      >
        Lorem, ipsum dolor
      </div>
    </div>
    <div
      class="mocked-styled-3 el-step-vertical"
    >
      <div
        class="mocked-styled-4 el-step-vertical-item"
      >
        <div
          class="mocked-styled-1 el-step"
          data-testid="step-1"
          role="button"
          tabindex="0"
        >
          2
        </div>
      </div>
      <div
        class="mocked-styled-5 el-step-vertical-content"
      >
        Lorem, ipsum dolor
      </div>
    </div>
    <div
      class="mocked-styled-3 el-step-vertical"
    >
      <div
        class="mocked-styled-4 el-step-vertical-item"
      >
        <div
          class="mocked-styled-1 el-step"
          data-testid="step-2"
          role="button"
          tabindex="0"
        >
          3
        </div>
      </div>
      <div
        class="mocked-styled-5 el-step-vertical-content"
      >
        Lorem, ipsum dolor
      </div>
    </div>
    <div
      class="mocked-styled-3 el-step-vertical"
    >
      <div
        class="mocked-styled-4 el-step-vertical-item"
      >
        <div
          class="mocked-styled-1 el-step"
          data-testid="step-3"
          role="button"
          tabindex="0"
        >
          4
        </div>
      </div>
      <div
        class="mocked-styled-5 el-step-vertical-content"
      >
        Lorem, ipsum dolor
      </div>
    </div>
    <div
      class="mocked-styled-3 el-step-vertical"
    >
      <div
        class="mocked-styled-4 el-step-vertical-item"
      >
        <div
          class="el-is-active el-step"
          data-testid="step-4"
          role="button"
          tabindex="0"
        >
          5
        </div>
      </div>
      <div
        class="mocked-styled-5 el-step-vertical-content"
      >
        Lorem, ipsum dolor
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`StepsVertical > should match a snapshot when selectedStep is different 1`] = `
<DocumentFragment>
  <div
    class="mocked-styled-2 el-steps-vertical"
  >
    <div
      class="mocked-styled-3 el-step-vertical"
    >
      <div
        class="mocked-styled-4 el-step-vertical-item"
      >
        <div
          class="mocked-styled-1 el-step"
          data-testid="step-0"
          role="button"
          tabindex="0"
        >
          1
        </div>
      </div>
      <div
        class="mocked-styled-5 el-step-vertical-content"
      >
        Lorem, ipsum dolor
      </div>
    </div>
    <div
      class="mocked-styled-3 el-step-vertical"
    >
      <div
        class="mocked-styled-4 el-step-vertical-item"
      >
        <div
          class="el-is-active el-step"
          data-testid="step-1"
          role="button"
          tabindex="0"
        >
          2
        </div>
      </div>
      <div
        class="mocked-styled-5 el-step-vertical-content"
      >
        Lorem, ipsum dolor
      </div>
    </div>
  </div>
</DocumentFragment>
`;
